### PR TITLE
JDK 11&17 Compatibility

### PR DESCRIPTION
**Pull request by nirmal101**
**Purpose**
Update the repository from mockito 1 to mockito 3 to support JDK 11 and JDK 17 compilation.

**When should this PR be merged**
After this PR is merged to the main branch, we can't build the branch on JDK 8

**Version checked,**

openjdk "11.0.16.1"
openjdk "17.0.4.1"

**Related Issue**
https://github.com/wso2/product-is/issues/14224

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml [hunk 1]
```diff
@@ -116,6 +121,12 @@
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.testutil</artifactId>
+            <version>5.23.40</version>
```
[the remark this answers is not in the document]
Comment (nirmal101) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml:127]: it is an unnecessary dependency. will remove it

File: components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml [hunk 2]
```diff
@@ -116,6 +121,7 @@
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
         </dependency>
+
```
[the remark this answers is not in the document]
Comment (nirmal101) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/pom.xml:124]: yes. will remove it.

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessorTest.java [hunk 3]
```diff
@@ -123,7 +125,7 @@ public class FederatedIdpInitLogoutProcessorTest extends PowerMockTestCase {
                     "EKgyYA0-FHSbFNRtk3-rN25biW3ivU5AWeo9W3dI6epcNSr4pCCvWBIKI-rk01J8kYyu2ZujecyD0yoz420lbZ2c_dMKFpCDH" +
                     "DdYjueK4tYE66jpAzvJEyPs37snH-6ok2YaoYjKudyfCdXni7Bg";
 
-    @BeforeTest
+    @BeforeMethod
```
[the remark this answers is not in the document]
Comment (nirmal101) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessorTest.java:128]: Due to the following exception.
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)

Found the solution [here.](
https://stackoverflow.com/questions/41956692/could-not-initialize-plugin-interface-org-mockito-plugins-mockmaker#:~:text=Solution%3A%20BeforeTest%20VS%20BeforeMethod)

**Comment by piraveena** [comment]
can you resolve the conflicts in pom file? the snapshot versions are different

**Review by piraveena — APPROVED** [review]

**Pull request merged**